docs/reference/plugins/python.rst

```diff
@@ -9,37 +9,61 @@ any of the following things:
   :file:`pyproject.toml` file.
 - Install packages using :command:`pip`.
 
```

Comment:
I think this is where we would describe the steps that the Plugin takes to build, to a reasonable degree. Here's an initial draft:

```
During the Build step, the plugin creates a virtual environment directly into ${CRAFT_PART_INSTALL}. Then, it uses pip to install needed Python packages as configured in the python-requirements, python-constraints and python-packages keywords. Next, if the source contains a setup.py or pyproject.toml file, the source itself is pip-installed.
```

I'm conflicted about how much we should talk about the whole symlink deal here. It feel like too much detail but at the same time it's a common source of bugs...